src/app-layer-smtp.c

```diff
@@ -1154,7 +1157,11 @@ static int SMTPParseCommandWithParam(SMTPState *state, uint8_t prefix_len, uint8
         return -1;
     memcpy(*target, state->current_line + i, spc_i - i);
     (*target)[spc_i - i] = '\0';
-    *target_len = spc_i - i;
+    if (spc_i - i > UINT16_MAX) {
```

Comment:
does this mean we're limiting lines to a max length now where previously we didn't?

Reply:
> does this mean we're limiting lines to a max length now where previously we didn't?

Eheh no it does not mean that

Previously (that is currently), we are already limiting lines to a max length

But as there is an implicit cast from 'int' to 'uint16_t', if the line length is 65536, it will turn to zero (instead of saturating it to `UINT16_MAX`)
```
app-layer-smtp.c:1157:25: warning: implicit conversion loses integer precision: 'int' to 'uint16_t' (aka 'unsigned short') [-Wimplicit-int-conversion]
    *target_len = spc_i - i;
```

So, should I rather increase the sizes of `uint16_t helo_len;` and such ?

Reply:
I see. No lets not allow for ridiculously long lines. But I think it should lead to another event if we reach the limit.

Reply:
Will do